src/prefect/settings.py

```diff
@@ -525,6 +545,14 @@ def default_cloud_ui_url(settings, value):
 can affect retry lengths.
 """
 
+PREFECT_CLIENT_RETRY_EXTRA_CODES = Setting(
+    str, default="", value_callback=status_codes_as_integers_in_range
+)
+"""
+A comma-separated list of extra HTTP status codes to retry on. Defaults to an empty string.
+429 and 503 are always retried.
+"""
```

Comment:
Let's add a warning here about retrying on other error codes — might cause unexpected behavior if the route is not idempotent.